combat/combat.cpp

```diff
@@ -101,7 +101,7 @@ bool Monster::updateCombat() {
     // If we're fighting a pet, see if we'll ignore the pet and attack a player instead
     if(target->isPet() && target->getMaster()) {
         if(target->inSameRoom(target->getMaster())) {
-            if(Random::get(1,100) < 20 || target->intelligence.getCur() >= 150 || target->flagIsSet(M_KILL_MASTER_NOT_PET)) {
+            if(Random::get(1,100) <= 20 || this->intelligence.getCur() >= 150 || this->flagIsSet(M_KILL_MASTER_NOT_PET)) {
```

Comment:
Could the intelligence check be removed? In general I feel like these kinds of checks are a little too nebulous e.g. the builder won't necessarily know the impact of setting a creature's intelligence, but they do know the impact of setting the `M_KILL_MASTER_NOT_PET` flag

Reply:
I dont see why not. For the most part, mobs DO have their intelligence set nowadays when we build them. A lot of the older ones still default to 100. At the least we can make it so that intelligence value is much higher than 150. Not many mobs are super intelligent, and 150, i admit, might be kind of low. I was just going with that because it was already that way.  I'll make it higher, like 220 or something.

Reply:
Hell, before I found that, it wasn't having any effect at all! Ran it by Bane. It's been a bug and he told me to go ahead and fix it. But I'll just make that value higher, because it stands to reason that a very intelligent mob would not need that flag set and would know better.

Reply:
The solution here isn't necessarily to remove all these nebulous checks, but to DOCUMENT them. That way builders actually DO know. Documentation has always been an ongoing problem. I intend to work on that a lot more soon, starting with things like this, because this is a very good point you are making. At least in the meantime, making the value higher shouldn't have such a huge effect on things as it could with it at 150.

Reply:
Intelligence has to be 220+ now, rather than 150.

Reply:
Did as I suggested in above comments. Intelligence has to be 220+ now. Documentation is ongoing.